### PR TITLE
Add `arch` to APT repository configuration

### PR DESCRIPTION
**Pull request by slovdahl**
On more recent Debian/Ubuntu versions, not specifying the `arch` gives a warning like this:

```
N: Skipping acquire of configured file 'main/binary-i386/Packages' as repository 'https://apt.envoyproxy.io jammy InRelease' doesn't support architecture 'i386'
```

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@slovdahl trying to understand if that message is happening because the binary is x86_64 - so i386 is not actually supported

**Comment by slovdahl** [comment]
That and combined with `dpkg --print-foreign-architectures` returning `i386`. In that case, this warning will be logged unless an APT repository is either 1) locked to just a specific arch using e.g. `[arch=amd64]` 2) the repository actually provides `i386` binaries.

On my Ubuntu 24.04 machine:
```
$ dpkg --print-foreign-architectures
i386
```

**Comment by phlax** [comment]
i see - you have i386 setup on a non-386 machine

i guess my hesitance to land this that we only provide x86_64 and arm64 so not sure it makes sense to insert a non-support arch, altho i guess in that case it would just fail and the problem should be fairly obvious

**Comment by phlax** [comment]
i can confirm locally i see the same issue

**Comment by phlax** [comment]
can you fix DCO and then im happy to land

**Comment by slovdahl** [comment]
Thanks! DCO fixed.

Yeah, I think it's quite common to have `i386` enabled, at least on desktop installations. I haven't done it intentionally at least.

And as you say, we could make it safer by failing faster on non-`amd64` and -`arm64` architectures, but I guess the end-result is the same with or without `arch=` set on unsupported architectures.

File: docs/root/start/install.rst [hunk 1]
```diff
@@ -20,31 +20,31 @@ Install Envoy on Debian-based Linux
...
+      $ echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bookworm main" | sudo tee /etc/apt/sources.list.d/envoy.list
       $ sudo apt-get update
       $ sudo apt-get install envoy
       $ envoy --version
 
    .. code-tab:: console Debian bullseye
 
       $ wget -O- https://apt.envoyproxy.io/signing.key | sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg
-      $ echo "deb [signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bullseye main" | sudo tee /etc/apt/sources.list.d/envoy.list
+      $ echo "deb [arch=$(dpkg --print-architecture), signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bullseye main" | sudo tee /etc/apt/sources.list.d/envoy.list
```
Comment (phlax) [inline docs/root/start/install.rst:31]:
```suggestion
      $ echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bullseye main" | sudo tee /etc/apt/sources.list.d/envoy.list
```

same elsewhere

Comment (slovdahl) [reply docs/root/start/install.rst]: Ouch, thanks, forgot that there were more than 1. Fixed.

**Review by phlax — APPROVED** [review]
lgtm, thanks for iterating @slovdahl

**Pull request merged**